### ISSUE TITLE
Opening bndb twice can get into a state where saving is impossible

### ISSUE THREAD

**Issue by yrp604** [Vector35/binaryninja-api#2945]
**Version and Platform (required):**
- Binary Ninja Version: 3248-dev
- OS: Windows
- OS Version: 11

**Bug Description:**
Opening the same bndb twice can lead to a state where saving changes is impossible

**Steps To Reproduce:**
Please provide all steps required to reproduce the behavior:
1. Open a bndb, make any change that marks the bndb as dirty
2. Open the same bndb again in a second tab
3. Close the entire binary ninja application, this should prompt a save dialog for the first bndb
4. At the prompt, hit save. This will fail because the database is locked from the second bndb
5. Close the unmodified bndb, this show not prompt any sort of save dialog
6. Attempt to save the changes made to the initial bndb, this should now be impossible

**Expected Behavior:**
I should always be able to save my changes.

Log from the above operations:
```
BNWriteDatabaseGlobal: database is locked
Could write history entries: BNWriteDatabaseGlobal
Error while saving database snapshot: database is locked
Could not load snapshot 58: Snapshot not found
Error while saving database snapshot: Current snapshot is null
```

**Comment by CouleeApps** [comment]
This is due to the save failing (database is locked) after the "current snapshot" variable has already been updated to point to the new snapshot. Since the save fails, the snapshot is never created (transaction reverted) and you get the aforementioned null error. 

I have a fix for this that should be coming out shortly, which reverts the current snapshot variable in the event of an error saving.

**Comment by CouleeApps** [comment]
This will be fixed in 3.0.3255-dev via the change previously mentioned.
